net/tcp/tcp_send.c

```diff
@@ -329,6 +332,20 @@ static void tcp_sendcommon(FAR struct net_driver_s *dev,
     }
 #endif /* CONFIG_NET_IPv4 */
 
+  /* If this is the very first connection, seqno and sndseq will be 0!
+   * Rational:
+   * We need to make tcp->seqno random because some servers will not accept
+   * it starting from zero. The way to do it is making the sndseq random
+   * because it will be "memcpyied" to tcp->seqno.
+   */
+
+  seqno = tcp_getsequence(tcp->seqno);
```

Comment:
why need change here? we can't update only tcp_initsequence instead

Reply:
Yeah. This point is also not clear to me

Reply:
Unfortunately just updating tcp_initsequence and calling it on tcp_connect() isn't enough because it is zeroed on other places.

Reply:
But it's better to call tcp_initsequence in the right place(connect/accept) instead hacking tcp_sendcommon

Reply:
This issue was difficult to track because with TCP_WRITE_BUFFER was enabled everything was working fine, just the random initialization was enough. The issue was happening because when TCP_WRITE_BUFFER is disabled the conn->rexmit_seq is copied to conn->sndseq in the tcp_timer.c file. Setting the rexmit with write value in the connect solved the issue.